packages/aws-cdk-lib/aws-ec2/test/vpc.test.ts

```diff
@@ -1622,6 +1625,112 @@ describe('vpc', () => {
 
     });
 
+    test('Can customize NAT instances V2 properties', () => {
+      // GIVEN
+      const stack = getTestStack();
+
+      // WHEN
+      const keyPair = KeyPair.fromKeyPairName(stack, 'KeyPair', 'KeyPairName');
+      const userData = UserData.forLinux();
+      userData.addCommands('echo "hello world!"');
+
+      const natGatewayProvider = NatProvider.instanceV2({
+        instanceType: InstanceType.of(InstanceClass.T3, InstanceSize.SMALL),
+        machineImage: new GenericLinuxImage({
+          'us-east-1': 'ami-1',
+        }),
+
+        creditSpecification: CpuCredits.UNLIMITED,
+        defaultAllowedTraffic: NatTrafficDirection.OUTBOUND_ONLY,
+        keyPair,
+        userData,
+
+        // Unusuable in its current state
+        // The VPC is required to create the security group,
+        // but the NAT Provider is required to create the VPC
+        // See https://github.com/aws/aws-cdk/issues/27527
+        // securityGroup,
```

Comment:
I can't see a possible use for `securityGroup`, it might be worth just deprecating it